### ISSUE TITLE
Fp with nonsense suggestion from type_repetition_in_bounds when using #[derivative]

### ISSUE THREAD

**Issue by jendrikw** [rust-lang/rust-clippy#9203]
### Summary

When using derivative to generate a Clone impl, clippy produces a FP. Also, the suggestion is nonsense:

`"I: Clone, I::Item: Clone": Iterator + "I: Clone, I::Item: Clone"`

### Lint Name

type_repetition_in_bounds

### Reproducer

I tried this code:

```rust
#![warn(clippy::type_repetition_in_bounds)]

use derivative::Derivative; // 2.2.0

// generates
// impl<I: Iterator> Clone for S<I> where I: Clone, I::Item: Clone { ... }
#[derive(Clone)]
pub struct S<I: Iterator> {
    pub i: I,
    pub item: I::Item,
}

#[derive(Derivative)]
#[derivative(
    Clone(bound = "I: Clone, I::Item: Clone"),
)]
pub struct Foo<I: Iterator> {
    inner: S<I>,
}
```

I saw this happen:

```
warning: this type has already been used as a bound predicate
  --> src/lib.rs:15:19
   |
15 |     Clone(bound = "I: Clone, I::Item: Clone"),
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> src/lib.rs:1:9
   |
1  | #![warn(clippy::type_repetition_in_bounds)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = help: consider combining the bounds: `"I: Clone, I::Item: Clone": Iterator + "I: Clone, I::Item: Clone"`
```

I expected to see this happen:

No warning.

### Version

```text
rustc 1.62.0 (a8314ef7d 2022-06-27)
binary: rustc
commit-hash: a8314ef7d0ec7b75c336af2c9857bfaf43002bfc
commit-date: 2022-06-27
host: x86_64-unknown-linux-gnu
release: 1.62.0
LLVM version: 14.0.5
```

### Additional Labels

@rustbot label +I-suggestion-causes-error
